doc/release-process.md

```diff
@@ -0,0 +1,14 @@
+# Release Process
+
+1. Open PR to master that
+   1. adds release notes to `doc/CHANGELOG.md` and
+   2. if this is **not** a patch release, updates `_PKG_VERSION_{MAJOR,MINOR}` and `_LIB_VERSIONS_*` in `configure.ac`
+2. After the PR is merged,
+   * if this is **not** a patch release, create a release branch with name `MAJOR.MINOR.x`.
```

Comment:
nit: So I assume `x` is a literal x here, right, so this is `1.2.x`. Then we could just call it `1.2`. I think this is in fact closer to Core (just `MAJOR.MINOR`), and this is how you usually abbreviate versions: Just drop the parts you don't care about.

Reply:
Core creates a MAJOR.x branch (where x is a literal x) according to @sipa's description ([core branches](https://github.com/bitcoin/bitcoin/branches)).

Reply:
I'm fine with either. I think Bitcoin Core went with "22.x" because branches that are just a number are a bit weird, but in the end it's all just arbitrary conventions.

Reply:
Ok, I removed the .x because it looks nicer.